pennylane/transforms/qmc.py

```diff
@@ -0,0 +1,353 @@
+        control_values=control_values,
+        work_wires=work_wires,
+    )
+
+    Hadamard(target_wire)
+    PauliX(target_wire)
+
+
+def _apply_controlled_v(target_wire, control_wire):
+    """Provides the circuit to apply a controlled version of the :math:`V` gate defined in
+    `this <https://arxiv.org/abs/1805.00109>`__ paper.
+
+    The :math:`V` gate is simply a Pauli-Z gate applied to the ``target_wire``, i.e., the ancilla
+    wire in which the expectation value is encoded.
+
+    The controlled version of this gate is then a CZ gate.
+
+    Args:
+        target_wire (Wires): the ancilla wire in which the expectation value is encoded
+        control_wire (Wires): the control wire from the register of phase estimation qubits
+    """
+    CZ(wires=[control_wire[0], target_wire[0]])
+
+
+def apply_controlled_Q(fn, wires, target_wire, control_wire, work_wires):
+    r"""Provides the circuit to apply a controlled version of the :math:`\mathcal{Q}` unitary
+    defined in `this <https://arxiv.org/abs/1805.00109>`__ paper.
+
+    Given a callable ``fn`` input corresponding to the :math:`\mathcal{F}` unitary in the above
+    paper, this function transforms the circuit into a controlled-version of the :math:`\mathcal{Q}`
+    unitary which forms part of the quantum Monte Carlo algorithm. In this algorithm, one of the
+    wires acted upon by :math:`\mathcal{F}`, specified by ``target_wire``, is used to embed a
+    Monte Carlo estimation problem. The :math:`\mathcal{Q}` unitary is then designed to encode the
+    target expectation value as a phase in one of its eigenvalues.
+
+    This function transforms to a controlled version of :math:`\mathcal{Q}` that is compatible with
+    quantum phase estimation (see :class:`~.QuantumPhaseEstimation` for more details).
+
+    Args:
+        fn (Callable): a quantum function that applies quantum operations according to the
+            :math:`\mathcal{F}` unitary used as part of quantum Monte Carlo estimation
+        wires (Union[Wires or Sequence[int]]): the wires acted upon by the ``fn`` circuit
+        target_wire (Union[Wires, int]): The wire in which the expectation value is encoded. Must be
+            contained within ``wires``.
+        control_wire (Union[Wires, int]): the control wire from the register of phase estimation
+            qubits
+        work_wires (Union[Wires, Sequence[int], or int]): additional work wires used when
+            decomposing :math:`\mathcal{Q}`
+
+    Returns:
+        function: The input function transformed to the :math:`\mathcal{Q}` unitary
+
+    Raises:
+        ValueError: if ``target_wire`` is not in ``wires``
+    """
+    fn_inv = adjoint(fn)
+
+    wires = Wires(wires)
+    target_wire = Wires(target_wire)
+    control_wire = Wires(control_wire)
+    work_wires = Wires(work_wires)
+
+    if not wires.contains_wires(target_wire):
+        raise ValueError("The target wire must be contained within wires")
+
+    @wraps(fn)
```

Comment:
It's defined [here](https://docs.python.org/3/library/functools.html#functools.wraps). The idea is to persist the name and docstring of the original function, rather than overwriting it with `wrapper`.

For example:
```python
from functools import wraps

def fn():
    """Docstring"""
    ...

def wrapper1():
    """Wrong Docstring"""
    fn()
    return 
    
@wraps(fn)
def wrapper2():
    """Wrong Docstring"""
    fn()
    return
```
The output is:
```pycon
>>> print("Name of wrapper1:     ", wrapper1.__name__)
Name of wrapper1:      wrapper1
>>> print("Docstring of wrapper1:", wrapper1.__doc__)
Docstring of wrapper1: Wrong Docstring
>>> print("Name of wrapper2:     ", wrapper2.__name__)
Name of wrapper2:      fn
>>> print("Docstring of wrapper2:", wrapper2.__doc__)
Docstring of wrapper2: Docstring
```

I'd never really used this before, so good chance for me to learn too!